fix task1

BodyanB opened the pull request:
Please view the code

BodyanB committed:
fix task1

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello World!");
+            string path = @"C:\Users\User\source\repos\altexsoft-lab-2020\ConsoleApp\ConsoleApp\bin\Debug\work.txt";
```
А что будет, если путь поменяется ?

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
+        static void ReadText(string path)
+        {
+            string Line;
+            string pathNew = @"C:\Users\User\source\repos\altexsoft-lab-2020\ConsoleApp\ConsoleApp\bin\Debug\note.txt";
```
А что будет, если путь поменяется ?

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
+
+                    Line = fs.ReadLine();
+
+                    var charsToRemove = new string[] { "@", ",", ".", ";", "'", "!", "?", "(", ")" };
```
Обратите внимание на регулярные выражения

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
+            string readText = File.ReadAllText(pathNew);
+            string pattern = @"\?\^\`\{\}\|";
+            int result = String.Compare(readText, pattern);
+            if (Regex.Matches(readText, pattern).Count == 0)
```
Проверку лучше делать до преобразований

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
+        {
+            string word;
+            string readText = File.ReadAllText(path);
+            string[] words = readText.Split(new char[] { ' ', '"', '.', '?', '!', ',', '(', ')' }, StringSplitOptions.RemoveEmptyEntries);
```
Обратите внимание на регулярные выражения

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
+        static void CountWord(string path)
+        {
+            string word;
+            string readText = File.ReadAllText(path);
```
Нет проверки на существование файла

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
+        }
+        static void ThirdSentence(string path)
+        {
+            using (StreamReader str = File.OpenText(path))
```
Нет проверки на существование файла

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
+        {
+            string Line;
+            string pathNew = @"C:\Users\User\source\repos\altexsoft-lab-2020\ConsoleApp\ConsoleApp\bin\Debug\note.txt";
+            using (StreamReader fs = File.OpenText(path))
```
Нет проверки на существование файла

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
+                int value = 0;
+                char[] charArr = new char[] { ' ' };
+                while (str.Peek() != -1)
+                {
```
Логику можно упростить

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
+            {
+                string[] dirs = Directory.GetDirectories(path, "*", SearchOption.TopDirectoryOnly);
+                Console.WriteLine("The number of directories is {0}.", dirs.Length);
+                for (int i = 0; i < dirs.Length; i++)
```
Нет сортировки

alex-geraschenko commented on ConsoleApp/Program.cs:
```diff
@@ -1,12 +1,177 @@
+
+                string[] fls = Directory.GetFiles(targetDirectory, "*");
+                Console.WriteLine("The number of files is {0}.", fls.Length);
+                foreach (string fl in fls)
```
Нет сортировки

alex-geraschenko reviewed (changes requested):
Постарайтесь разнести методы задач по разным классам в соответствии с принципами SOLID

BodyanB committed:
make a path

BodyanB committed:
fix task1

BodyanB committed:
commit 12.08

alex-geraschenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,175 @@
+    class Menu
+    {
+        #region Private Members
+        private List<string> m_datafromfile = new List<string>();
```
В C# приватные поля принято называть начиная с `_` без префикса `m`

alex-geraschenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,175 @@
+        #endregion
+
+        #region Public Properties
+        public List<string> MAINMENUFILEDATA
```
В C# публичные свойства принято называться в `PascalCase`

alex-geraschenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,175 @@
+            }
+
+        }
+        public string[] FILEPATHS
```
В C# публичные свойства принято называться в `PascalCase`

alex-geraschenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,175 @@
+        #region Public Properties
+        public List<string> MAINMENUFILEDATA
+        {
+            get
```
Используйте `Auto Property`

alex-geraschenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,175 @@
+        }
+        public string[] FILEPATHS
+        {
+            get
```
Используйте `Auto Property`

alex-geraschenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,175 @@
+
+        #region Public Member Functions
+
+        private void readRelaventFile()
```
В C# все методы принято называть в `PascalCase`

alex-geraschenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,175 @@
+
+        }
+
+        public void showMenuScreen()
```
В C# все методы принято называть в `PascalCase`

alex-geraschenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,175 @@
+            {
+                MAINMENUFILEDATA.Insert(1, autndate);
+            }
+        top: foreach (string item in MAINMENUFILEDATA)
```
Использование меток и `goto` плохая практика. Постарайтесь найти другое решение

alex-geraschenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,175 @@
+        private void ReadDataFromMainFile()
+        {
+            string path = FILEPATHS.First(p => p.Contains("Main"));
+            StreamReader srObj = new StreamReader(path);
```
Изучите, что такое `IDisposable`, `Dispose` и конструкция `using`

alex-geraschenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,175 @@
+
+        }
+
+        public void readFilesPath()
```
В C# все методы принято называть в `PascalCase`

alex-geraschenko commented on ConsoleApp/ReadPath.cs:
```diff
@@ -0,0 +1,25 @@
+
+namespace ConsoleApp
+{
+    internal class ReadFile
```
Классы принято называть в существительными

alex-geraschenko commented on ConsoleApp/ReadText.cs:
```diff
@@ -0,0 +1,55 @@
+
+namespace ConsoleApp
+{
+    class ReadText : ReadFile
```
Классы принято называть в существительными

alex-geraschenko commented on ConsoleApp/CountWord.cs:
```diff
@@ -0,0 +1,29 @@
+
+namespace ConsoleApp
+{
+    class CountWord : ReadText
```
Классы принято называть в существительными

BodyanB committed:
new version 1.01

serhii-shkodenko commented on ConsoleApp/FolderName.cs:
```diff
@@ -0,0 +1,54 @@
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("The process failed: {0}", e.ToString());
```
Вы вызываете ToString() на объекте, а Вам нужен текст исключения.

serhii-shkodenko commented on ConsoleApp/FolderName.cs:
```diff
@@ -0,0 +1,54 @@
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("The process failed: {0}", e.ToString());
```
Вы вызываете ToString() на объекте, а Вам нужен текст исключения.

serhii-shkodenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,70 @@
+{
+    class Menu 
+    {
+        static public string Comand;
```
В C# принято указываеть модификаторы доступа в самом начале, перед всеми остальными модификаторами. 
https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference?view=vs-2017
csharp_preferred_modifier_order = public,private,protected,internal,static,extern,new,virtual,abstract,sealed,override,readonly,unsafe,volatile,async:silent

serhii-shkodenko replied:
Вы используете переменную `Comand` только внтури метода. Раз она Вам нужна только там, то и опеределяйте ее там.

serhii-shkodenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,70 @@
+
+        public void ShowMenuScreen()
+        {
+            string autndate = "Author: B/B\t\t\t\t\t  Date:" + System.DateTime.Now;
```
Локальные переменные следует именовать в `camelCase`.

serhii-shkodenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,70 @@
+                if (Comand == "5") break;
+                Console.WriteLine(autndate);
+                Console.WriteLine("### MENU ###");
+                Console.WriteLine("1. Считать по указанному в пути текстовый файл и удалить в нем указанный в консоли символ/слово.");
```
Вот так выглядит у меня меню на моей машине с английской локалью: 
![encoding_error](https://user-images.githubusercontent.com/69808191/90545620-9389a380-e191-11ea-8377-6649e3a1117d.png)

Пожалуйста, установите необходимую кодировку для консоли.

serhii-shkodenko commented on ConsoleApp/Menu.cs:
```diff
@@ -0,0 +1,70 @@
+            {
+
+                if (Comand == "5") break;
+                Console.WriteLine(autndate);
```
Задумайтесь, что будет если пользователь не введет ни один из перечисленных Вами вариантов. Например, будет повторяться одно и то же время.

serhii-shkodenko commented on ConsoleApp/FolderName.cs:
```diff
@@ -0,0 +1,54 @@
+        {
+            Console.WriteLine(@"Enter the path:");
+            string path = Console.ReadLine();
+            Dictionary<Int32, string> dirеctory = new Dictionary<int, string>();
```
Следует использовать сокращеннное наименование типа.

serhii-shkodenko commented on ConsoleApp/FolderName.cs:
```diff
@@ -0,0 +1,54 @@
+{
+    class FolderName
+    {
+        public void FName()
```
В С# следует использовать глаголы для имен методов
https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/4df752aw(v=vs.71)

serhii-shkodenko commented on ConsoleApp/Path.cs:
```diff
@@ -0,0 +1,25 @@
+
+namespace ConsoleApp
+{
+    internal class ReadFile
```
В своем коде Вы наследуетесь от этого класса. Если абстрагироваться от текущей реализации, очевидно, этот класс предназначен для инкапсуляции логики чтения файла. Вполне вероятно, что Вам понадобится использовать этот механизм в разных частях Вашей программы. Поэтому кажется неразумным "запирать" его в рамках одной библиотеки.

BodyanB replied:
В этом месте не понял что именно поправить, можно уточнить?

serhii-shkodenko commented on ConsoleApp/Text.cs:
```diff
@@ -0,0 +1,55 @@
+            string textStream = ReadT();
+            Console.Write("Введите слово или символ для удаления. ");
+            string word = Console.ReadLine();
+            if (String.IsNullOrEmpty(word))
```
Следует использовать более простое наименование типа.

serhii-shkodenko commented on ConsoleApp/Path.cs:
```diff
@@ -0,0 +1,25 @@
+
+namespace ConsoleApp
+{
+    internal class ReadFile
```
Имя файла принято называть так же, как и имя класса

serhii-shkodenko commented on ConsoleApp/Text.cs:
```diff
@@ -0,0 +1,55 @@
+                {
+                    string textReplace = textStream.Replace(word, "");
+                    Console.WriteLine(textReplace);
+                    Console.Write("Введите путь для сохранения файла: ");
```
Вы выделили логику чтения файла в отдельный класс, почему не поступили так же с логикой записи?

serhii-shkodenko commented on ConsoleApp/Text.cs:
```diff
@@ -0,0 +1,55 @@
+                    Console.WriteLine(textReplace);
+                    Console.Write("Введите путь для сохранения файла: ");
+                    string writePath = (Console.ReadLine());
+                    string writePathFile = writePath + "note.txt";
```
Я бы вынес название файла в константу.

serhii-shkodenko commented on ConsoleApp/ThirdSentence.cs:
```diff
@@ -0,0 +1,35 @@
+{
+    class ThirdSentence : Text
+    {
+        public void TSentence()           
```
Название метода должно быть в глагольной форме
https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/4df752aw(v=vs.71)

serhii-shkodenko commented on ConsoleApp/ThirdSentence.cs:
```diff
@@ -0,0 +1,35 @@
+                    reverseWords.Add(new string(word.ToCharArray().Reverse().ToArray()));
+                }
+
+                Console.WriteLine(String.Join(' ', reverseWords));
```
Просто `string`

serhii-shkodenko commented on ConsoleApp/Word.cs:
```diff
@@ -0,0 +1,29 @@
+{
+    class Word : Text
+    {
+        public void CWord()
```
Приведите название метода в глагольную форму. 
https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/4df752aw(v=vs.71)

serhii-shkodenko commented on ConsoleApp/Word.cs:
```diff
@@ -0,0 +1,29 @@
+            
+            string word;
+            string textStream = ReadT();
+            string[] words = textStream.Split(new char[] { ' ', '"', '.', '?', '!', ',', '(', ')' }, StringSplitOptions.RemoveEmptyEntries);
```
Используйте регулярные выражения.

serhii-shkodenko commented on ConsoleApp/Word.cs:
```diff
@@ -0,0 +1,29 @@
+                word = words[i];
+                resultWords.Add(word);
+            }
+            Console.WriteLine(String.Join(',', resultWords));
```
Упростите используемое имя типа.

BodyanB commented:
Доброе утро Сергей. Я так понимаю что теперь с Вами работаю?

вт, 18 авг. 2020 г. в 23:09, Sergiy <notifications@github.com>:

> *@serhii-shkodenko* requested changes on this pull request.
> ------------------------------
>
> In ConsoleApp/FolderName.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472348424>
> :
>
> > +            Dictionary<Int32, string> dirеctory = new Dictionary<int, string>();
> +            try
> +            {
> +                string[] dirs = Directory.GetDirectories(path, "*", SearchOption.TopDirectoryOnly);
> +                Array.Sort(dirs);
> +                Console.WriteLine("The number of directories is {0}.", dirs.Length);
> +                for (int i = 0; i < dirs.Length; i++)
> +                {
> +                    dirеctory.Add(i + 1, dirs[i]);
> +                    Console.WriteLine($" {i + 1}.{dirs[i]}");
> +                }
> +
> +            }
> +            catch (Exception e)
> +            {
> +                Console.WriteLine("The process failed: {0}", e.ToString());
>
> Вы вызываете ToString() на объекте, а Вам нужен текст исключения.
> ------------------------------
>
> In ConsoleApp/FolderName.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472348913>
> :
>
> > +
> +
> +            try
> +            {
> +
> +                string[] fls = Directory.GetFiles(targetDirectory, "*");
> +                Array.Sort(fls);
> +                Console.WriteLine("The number of files is {0}.", fls.Length);
> +                foreach (string fl in fls)
> +                {
> +                    Console.WriteLine(fl);
> +                }
> +            }
> +            catch (Exception e)
> +            {
> +                Console.WriteLine("The process failed: {0}", e.ToString());
>
> Вы вызываете ToString() на объекте, а Вам нужен текст исключения.
> ------------------------------
>
> In ConsoleApp/Menu.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472356093>
> :
>
> > @@ -0,0 +1,70 @@
> +﻿using System;
> +using System.Collections.Generic;
> +using System.Text;
> +using System.Linq;
> +using System.Threading.Tasks;
> +using System.IO;
> +using System.Threading;
> +
> +namespace ConsoleApp
> +{
> +    class Menu
> +    {
> +        static public string Comand;
>
> В C# принято указываеть модификаторы доступа в самом начале, перед всеми
> остальными модификаторами.
>
> https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference?view=vs-2017
> csharp_preferred_modifier_order =
> public,private,protected,internal,static,extern,new,virtual,abstract,sealed,override,readonly,unsafe,volatile,async:silent
> ------------------------------
>
> In ConsoleApp/Menu.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472358896>
> :
>
> > +using System.Collections.Generic;
> +using System.Text;
> +using System.Linq;
> +using System.Threading.Tasks;
> +using System.IO;
> +using System.Threading;
> +
> +namespace ConsoleApp
> +{
> +    class Menu
> +    {
> +        static public string Comand;
> +
> +        public void ShowMenuScreen()
> +        {
> +            string autndate = "Author: B/B\t\t\t\t\t  Date:" + System.DateTime.Now;
>
> Локальные переменные следует именовать в camelCase.
> ------------------------------
>
> In ConsoleApp/Menu.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472364588>
> :
>
> > +{
> +    class Menu
> +    {
> +        static public string Comand;
> +
> +        public void ShowMenuScreen()
> +        {
> +            string autndate = "Author: B/B\t\t\t\t\t  Date:" + System.DateTime.Now;
> +
> +            while (true)
> +            {
> +
> +                if (Comand == "5") break;
> +                Console.WriteLine(autndate);
> +                Console.WriteLine("### MENU ###");
> +                Console.WriteLine("1. Считать по указанному в пути текстовый файл и удалить в нем указанный в консоли символ/слово.");
>
> Вот так выглядит у меня меню на моей машине с английской локалью:
> [image: encoding_error]
> <https://user-images.githubusercontent.com/69808191/90545620-9389a380-e191-11ea-8377-6649e3a1117d.png>
>
> Пожалуйста, установите необходимую кодировку для консоли.
> ------------------------------
>
> In ConsoleApp/Menu.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472404270>
> :
>
> > +
> +namespace ConsoleApp
> +{
> +    class Menu
> +    {
> +        static public string Comand;
> +
> +        public void ShowMenuScreen()
> +        {
> +            string autndate = "Author: B/B\t\t\t\t\t  Date:" + System.DateTime.Now;
> +
> +            while (true)
> +            {
> +
> +                if (Comand == "5") break;
> +                Console.WriteLine(autndate);
>
> Задумайтесь, что будет если пользователь не введет ни один из
> перечисленных Вами вариантов. Например, будет повторяться одно и то же
> время.
> ------------------------------
>
> In ConsoleApp/Menu.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472407384>
> :
>
> > @@ -0,0 +1,70 @@
> +﻿using System;
> +using System.Collections.Generic;
> +using System.Text;
> +using System.Linq;
> +using System.Threading.Tasks;
> +using System.IO;
> +using System.Threading;
> +
> +namespace ConsoleApp
> +{
> +    class Menu
> +    {
> +        static public string Comand;
>
> Вы используете переменную Comand только внтури метода. Раз она Вам нужна
> только там, то и опеределяйте ее там.
> ------------------------------
>
> In ConsoleApp/FolderName.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472421681>
> :
>
> > @@ -0,0 +1,54 @@
> +﻿using System;
> +using System.Collections.Generic;
> +using System.IO;
> +using System.Text;
> +
> +namespace ConsoleApp
> +{
> +    class FolderName
> +    {
> +        public void FName()
> +        {
> +            Console.WriteLine(@"Enter the path:");
> +            string path = Console.ReadLine();
> +            Dictionary<Int32, string> dirеctory = new Dictionary<int, string>();
>
> Следует использовать сокращеннное наименование типа.
> ------------------------------
>
> In ConsoleApp/FolderName.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472422619>
> :
>
> > @@ -0,0 +1,54 @@
> +﻿using System;
> +using System.Collections.Generic;
> +using System.IO;
> +using System.Text;
> +
> +namespace ConsoleApp
> +{
> +    class FolderName
> +    {
> +        public void FName()
>
> В С# следует использовать глаголы для имен методов
>
> https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/4df752aw(v=vs.71)
> ------------------------------
>
> In ConsoleApp/Path.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472428694>
> :
>
> > @@ -0,0 +1,25 @@
> +﻿using System;
> +using System.IO;
> +
> +namespace ConsoleApp
> +{
> +    internal class ReadFile
>
> В своем коде Вы наследуетесь от этого класса. Если абстрагироваться от
> текущей реализации, очевидно, этот класс предназначен для инкапсуляции
> логики чтения файла. Вполне вероятно, что Вам понадобится использовать этот
> механизм в разных частях Вашей программы. Поэтому кажется неразумным
> "запирать" его в рамках одной библиотеки.
> ------------------------------
>
> In ConsoleApp/Text.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472432795>
> :
>
> > +using System.Text;
> +using System.IO;
> +using System.Linq;
> +using System.Text.RegularExpressions;
> +
> +namespace ConsoleApp
> +{
> +    class Text : ReadFile
> +    {
> +
> +        public void RText()
> +        {
> +            string textStream = ReadT();
> +            Console.Write("Введите слово или символ для удаления. ");
> +            string word = Console.ReadLine();
> +            if (String.IsNullOrEmpty(word))
>
> Следует использовать более простое наименование типа.
> ------------------------------
>
> In ConsoleApp/Path.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472436948>
> :
>
> > @@ -0,0 +1,25 @@
> +﻿using System;
> +using System.IO;
> +
> +namespace ConsoleApp
> +{
> +    internal class ReadFile
>
> Класс у Вас называется ReadFile, а файл Path.cs. Нехорошо.
> ------------------------------
>
> In ConsoleApp/Text.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472437444>
> :
>
> > +
> +        public void RText()
> +        {
> +            string textStream = ReadT();
> +            Console.Write("Введите слово или символ для удаления. ");
> +            string word = Console.ReadLine();
> +            if (String.IsNullOrEmpty(word))
> +            {
> +                Console.WriteLine(textStream);
> +            }
> +            else
> +               if (textStream.Contains(word))
> +                {
> +                    string textReplace = textStream.Replace(word, "");
> +                    Console.WriteLine(textReplace);
> +                    Console.Write("Введите путь для сохранения файла: ");
>
> Вы выделили логику чтения файла в отдельный класс, почему не поступили так
> же с логикой записи?
> ------------------------------
>
> In ConsoleApp/Text.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472438648>
> :
>
> > +        {
> +            string textStream = ReadT();
> +            Console.Write("Введите слово или символ для удаления. ");
> +            string word = Console.ReadLine();
> +            if (String.IsNullOrEmpty(word))
> +            {
> +                Console.WriteLine(textStream);
> +            }
> +            else
> +               if (textStream.Contains(word))
> +                {
> +                    string textReplace = textStream.Replace(word, "");
> +                    Console.WriteLine(textReplace);
> +                    Console.Write("Введите путь для сохранения файла: ");
> +                    string writePath = (Console.ReadLine());
> +                    string writePathFile = writePath + "note.txt";
>
> Я бы вынес название файла в константу.
> ------------------------------
>
> In ConsoleApp/ThirdSentence.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472441985>
> :
>
> > @@ -0,0 +1,35 @@
> +﻿using System;
> +using System.Collections.Generic;
> +using System.Linq;
> +using System.IO;
> +
> +namespace ConsoleApp
> +{
> +    class ThirdSentence : Text
> +    {
> +        public void TSentence()
>
> Название метода должно быть в глагольной форме
>
> https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/4df752aw(v=vs.71)
> ------------------------------
>
> In ConsoleApp/ThirdSentence.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472444096>
> :
>
> > +            string textStream = ReadT();
> +
> +            string[] sentences = textStream.Split(new char[] { '.' }, StringSplitOptions.RemoveEmptyEntries);
> +            if (sentences.Length >= 3)
> +            {
> +                string senrence = sentences[2];
> +                string[] words = senrence.Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
> +                string word;
> +                List<string> reverseWords = new List<string>();
> +                for (int i = 0; i < words.Length; i++)
> +                {
> +                    word = words[i];
> +                    reverseWords.Add(new string(word.ToCharArray().Reverse().ToArray()));
> +                }
> +
> +                Console.WriteLine(String.Join(' ', reverseWords));
>
> Просто string
> ------------------------------
>
> In ConsoleApp/Word.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472445653>
> :
>
> > @@ -0,0 +1,29 @@
> +﻿using System;
> +using System.Collections.Generic;
> +using System.Text;
> +using System.IO;
> +
> +namespace ConsoleApp
> +{
> +    class Word : Text
> +    {
> +        public void CWord()
>
> Приведите название метода в глагольную форму.
>
> https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/4df752aw(v=vs.71)
> ------------------------------
>
> In ConsoleApp/Word.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472446084>
> :
>
> > @@ -0,0 +1,29 @@
> +﻿using System;
> +using System.Collections.Generic;
> +using System.Text;
> +using System.IO;
> +
> +namespace ConsoleApp
> +{
> +    class Word : Text
> +    {
> +        public void CWord()
> +        {
> +
> +            string word;
> +            string textStream = ReadT();
> +            string[] words = textStream.Split(new char[] { ' ', '"', '.', '?', '!', ',', '(', ')' }, StringSplitOptions.RemoveEmptyEntries);
>
> Используйте регулярные выражения.
> ------------------------------
>
> In ConsoleApp/Word.cs
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#discussion_r472447096>
> :
>
> > +        public void CWord()
> +        {
> +
> +            string word;
> +            string textStream = ReadT();
> +            string[] words = textStream.Split(new char[] { ' ', '"', '.', '?', '!', ',', '(', ')' }, StringSplitOptions.RemoveEmptyEntries);
> +
> +            Console.WriteLine("Количество слов:" + words.Length);
> +
> +            List<string> resultWords = new List<string>();
> +            for (int i = 9; i < words.Length; i += 10)
> +            {
> +                word = words[i];
> +                resultWords.Add(word);
> +            }
> +            Console.WriteLine(String.Join(',', resultWords));
>
> Упростите используемое имя типа.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#pullrequestreview-469654412>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFUJLDKRRB3FIGHPJGJYPGDSBLNXFANCNFSM4PUIWR5A>
> .
>

serhii-shkodenko commented:
> Доброе утро Сергей. Я так понимаю что теперь с Вами работаю?

Вы работаете со всеми менторами. Всех участников курсов будут ревьююить либо кто-то из нас, либа оба ментора.

BodyanB committed:
Notes fixed.  Programm v.1.0.2

alex-geraschenko commented:
У Вас еще не все комментарии исправлены, поэтому делать ревью еще рано

BodyanB commented:
Добрый день. Какие комментарии исправить?

чт, 20 авг. 2020 г. в 12:23, Alex Geraschenko <notifications@github.com>:

> У Вас еще не все комментарии исправлены, поэтому делать ревью еще рано
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#issuecomment-677485345>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFUJLDOC6PDGEF44UZMERPDSBTTSVANCNFSM4PUIWR5A>
> .
>

serhii-shkodenko commented:
Те замечания, которые Вы исправили в своем последнем коммите, пожалуйста, пометьте исправленными через кнопку Resolve Conversation. Кроме того, последний Ваш коммит не содержит всех исправлений, насколько я успел его просмотреть.

BodyanB commented:
Все замечания были исправлены.

чт, 20 авг. 2020 г. в 13:26, Sergiy <notifications@github.com>:

> Те замечания, которые Вы исправили в своем последнем коммите, пожалуйста,
> пометьте исправленными через кнопку Resolve Conversation. Кроме того,
> последний Ваш коммит не содержит всех исправлений, насколько я успел его
> прсмотреть.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#issuecomment-677514129>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFUJLDN6VK22VIMXROCGQBTSBT263ANCNFSM4PUIWR5A>
> .
>

alex-geraschenko commented:
Пересмотрите еще раз все замечания, которые есть на вкладке Conversation, и убедитесь, что все замечания поняты и исправлены.

BodyanB commented:
Хорошо.

On Fri, Aug 21, 2020, 22:26 Alex Geraschenko <notifications@github.com>
wrote:

> Пересмотрите еще раз все замечания, которые есть на вкладке Conversation,
> и убедитесь, что все замечания поняты и исправлены.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#issuecomment-678453860>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFUJLDJ2FPE5CK45L3HHM7DSB3C7DANCNFSM4PUIWR5A>
> .
>

BodyanB commented:
Добрый вечер. Все комментарии исправлены, либо переписаны.

alex-geraschenko commented:
> Добрый вечер. Все комментарии исправлены, либо переписаны.

Вы уверены? А если зайти на вкладку **Files changed** и посмотреть там?

BodyanB committed:
version 1.04 files changed

BodyanB commented:
Добрый день Александр. Те замечания что в  *Files changed* тоже поправил.

вт, 25 авг. 2020 г. в 22:05, Alex Geraschenko <notifications@github.com>:

> Добрый вечер. Все комментарии исправлены, либо переписаны.
>
> Вы уверены? А если зайти на вкладку *Files changed* и посмотреть там?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#issuecomment-680214869>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFUJLDLXJRBTADRO4GQSR5LSCQDPRANCNFSM4PUIWR5A>
> .
>

alex-geraschenko reviewed (approved):
Есть нюансы в именовании классов - из имени не понятно, для чего нужен класс. Для этого надо почитать Боба Мартина "Чистый Код".  Но в целом пойдет. Можно мержить и переходить ко второму заданию

BodyanB commented:
Спасибо.

ср, 26 авг. 2020 г. в 19:44, Alex Geraschenko <notifications@github.com>:

> *@alex-geraschenko* approved this pull request.
>
> Есть нюансы в именовании классов - из имени не понятно, для чего нужен
> класс. Для этого надо почитать Боба Мартина "Чистый Код". Но в целом
> пойдет. Можно мержить и переходить ко второму заданию
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BodyanB/altexsoft-lab-2020/pull/1#pullrequestreview-475672222>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFUJLDOROYSDB45L423V573SCU3YRANCNFSM4PUIWR5A>
> .
>
